codegen/java/src/main/kotlin/io/spine/protodata/codegen/java/file/BeforePrimaryDeclaration.kt

```diff
@@ -32,15 +32,24 @@ import io.spine.protodata.renderer.LineNumber
 import io.spine.protodata.renderer.LineNumber.Companion.notInFile
 import java.lang.System.lineSeparator
 
-private val pattern = Regex(
-    "(public\\s+)?(final\\s+)?(abstract\\s+)?((class)|(@?interface)|(enum))\\s+"
-)
+/**
+ * A pattern matching a top-level Java declaration.
+ *
+ * This pattern matches a type declaration, such as a class, an interface, an annotation
```

Comment:
I would have a comma after "an annotation" as well. Just like you do in the similar sentence down below.